ocaml/database/metastore/config_field.ml

```diff
@@ -0,0 +1,74 @@
+type nonrec ('a, 'b) result = ('a, ([> `Msg of string] as 'b)) result
+
+let ( let+ ) = Result.bind
+
+type positive = int64 [@@deriving rpcty]
+
+let positive i64 =
+  if Int64.compare i64 Int64.zero > 0 then
+    Ok i64
+  else
+    Fmt.error_msg "Value must be > 0: %a" Fmt.int64 i64
+
+type limit = int64 [@@deriving rpcty]
+
+let positive_or_unlimited = function None -> Ok 0L | Some i64 -> positive i64
+
+type milliseconds = int64 [@@deriving rpcty]
+
+let milliseconds f = f *. 1000. |> Float.round |> Int64.of_float |> positive
+
+type uri = Uri.t
+
+let typ_of_uri =
+  Serialization.using ~aname:"uri" Uri.to_string Uri.of_string
+    Rpc.Types.string.Rpc.Types.ty
+
+let uri ~ip ~port =
+  if port <= 0 then
+    Fmt.error_msg "Port must be > 0: %d" port
+  else
+    Uri.make ~host:(Ipaddr.to_string ip) ~port () |> Result.ok
+
+type path = Fpath.t
+
+let typ_of_path =
+  Serialization.using ~aname:"path" Fpath.to_string Fpath.v
+    Rpc.Types.string.Rpc.Types.ty
+
+type initial_cluster_state = New | Existing [@@deriving rpcty]
+
+type log_output = Stdout | Stderr | Default [@@deriving rpcty]
+
+type level = CRITICAL | ERROR | WARNING | INFO | DEBUG [@@deriving rpcty]
```

Comment:
Can't this be taken from the Logs or xapi-log libraries? It's already being duplicated

Reply:
This is the list of values accepted by 'etcd', I'll add a comment